packages/config-array/src/config-array.js

```diff
@@ -490,6 +490,67 @@ function assertExtraConfigTypes(extraConfigTypes) {
 	}
 }
 
+/**
+ * Calculates whether a file has a config or why it doesn't.
+ * @param {ConfigArray} configArray A normalized config array.
+ * @param {string} filePath The path of the file to check.
+ * @returns {"ignored"|"external"|"unconfigured"|"matched"} One of the constants returned by
+ * {@linkcode ConfigArray.getConfigStatus}.
+ */
+function calculateConfigStatus(configArray, filePath) {
```

Comment:
There's a lot of repeated logic from `getConfig`, and a separate cache.

Maybe we could rename the current  `getConfig` to `getConfigWithStatus`, and modify it to cache and return `{ config, status }`? Then, the new `getConfig` would be just `return this.getConfigWithStatus(filePath).config;`, and `getConfigStatus` would be just `return this.getConfigWithStatus(filePath).status;`.

Reply:
Yeah, I also thought that. Would like @nzakas' perspective on this.

Reply:
I think that makes sense. :+1:

Reply:
Thanks for the suggestion, done in 5ce6b188226bdc1d569723d78175cd5e2aff4cda. Please, have a look.